### ISSUE TITLE
Enable specifying a Markdown description of a category

### ISSUE THREAD

**Issue by parlough** [dart-lang/dartdoc#3618]
Then we can add descriptions on the API docs overview/index page: https://api.dart.dev/main

This would allow us to explain what the categories actually mean and why they exist. For example:

- That "Core" libraries should work across all platforms
- That "VM" libraries only work when ran or compiled for native platforms
- That usage of "WEB (LEGACY)" libraries should be migrated (and what to)
- Perhaps more :) 

While a normal string would be sufficient, supporting Markdown here would be great as it would enable easy links.

**Comment by parlough** [comment]
\cc @devoncarew Since you're modifying the categories now, what do you think?

This could show up here on the index page:

<img width="585" alt="New legacy category" src="https://github.com/dart-lang/dartdoc/assets/18372958/484f6cad-4762-473a-a309-f707b448d59a">

**Comment by devoncarew** [comment]
Yeah, I think this would be great - it would give more context to the categories.

We can manually describe things in the markdown above the generated list, but messaging things more than once can be useful.

**Comment by srawlins** [comment]
I think I got mixed up, but this is still not implemented, right? I think @devoncarew landed something separate, which enabled `package:web` to be listed in the Web category of the Dart SDK, right?
